browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -538,6 +545,74 @@ void RewardsDOMHandler::Init() {
+
+  // Configure a pref change registrar to update brave://rewards when settings
+  // are changed via brave://settings
+  InitPrefChangeRegistrar();
+}
+
+void RewardsDOMHandler::InitPrefChangeRegistrar() {
+  Profile* profile = Profile::FromWebUI(web_ui());
+  pref_change_registrar_.Init(profile->GetPrefs());
+
+  pref_change_registrar_.Add(
+      ads::prefs::kEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      ads::prefs::kAdsPerHour,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      ads::prefs::kAdsSubdivisionTargetingCode,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAutoContributeEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAutoContributeAmount,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kMinVisitTime,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kMinVisits,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAllowNonVerified,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kAllowVideoContribution,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipTwitterEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipRedditEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+  pref_change_registrar_.Add(
+      brave_rewards::prefs::kInlineTipGithubEnabled,
+      base::BindRepeating(&RewardsDOMHandler::OnPreferenceChanged,
+                          base::Unretained(this)));
+}
+
+void RewardsDOMHandler::OnPreferenceChanged(const std::string& key) {
+  AllowJavascript();
+  GetEnabledInlineTippingPlatforms(base::Value::ConstListView());
+  GetAutoContributionAmount(base::Value::ConstListView());
```

Comment:
By calling these message handlers directly, we're more-or-less acting on behalf of the front-end code. I don't have a strong opinion on that, but consider the tradeoffs of this versus the alternative that has the front-end listen for a generic "pref change" message and trigger these calls itself.